### PR TITLE
Update Goodies to produce Java 17 binaries

### PR DESCRIPTION
**Pull request by mpiggott**
* Produce Java 17 binaries
* Replace usages of URLConnection with Java's builtin http client
* Updated keystores with current algorithms 
* Removed matchers unused Nexus which required meaningful updates

### CODE REVIEW

File: httpfixture/src/test/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyProxyProviderTest.java [hunk 1]
```diff
@@ -12,81 +12,52 @@
...
 public class JettyProxyProviderTest
     extends TestSupport
 {
-  @Before
-  @IgnoreJRERequirement
-  public void clearJvmHttpAuthCaches() {
-    // FIXME: this is circumvention got from here: http://stackoverflow.com/questions/480895/reset-the-authenticator-credentials
-    // Actual Java bug is this: http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=6626700
-    sun.net.www.protocol.http.AuthCacheValue.setAuthCache(new sun.net.www.protocol.http.AuthCacheImpl());
-  }
```
Comment (mpiggott) [inline httpfixture/src/test/java/org/sonatype/goodies/httpfixture/server/jetty/impl/JettyProxyProviderTest.java:52]: Removing all the usage of Java internals in favour of the `java.net.http.HttpClient`

File: pom.xml [hunk 2]
```diff
@@ -233,31 +234,15 @@
       </dependency>
 
       <dependency>
-        <groupId>com.github.tomakehurst</groupId>
-        <artifactId>wiremock</artifactId>
-        <version>1.38</version> <!-- Later versions depend on Jackson 2.4, so we'd need to update too -->
```
Comment (mpiggott) [inline pom.xml:238]: I didn't find an easy upgrade path and when I checked Nexus 3 we don't appear to use the related matchers so I removed them.

File: pom.xml [hunk 3]
```diff
@@ -233,31 +234,15 @@
...
-        </exclusions>
+        <groupId>javax.xml.bind</groupId>
+        <artifactId>jaxb-api</artifactId>
+        <version>2.3.1</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.glassfish.jaxb</groupId>
+        <artifactId>jaxb-runtime</artifactId>
+        <version>2.3.9</version>
```
Comment (mpiggott) [inline pom.xml:245]: jaxb was removed in a recent JDK

File: testsupport/src/test/java/org/sonatype/goodies/testsupport/hamcrest/FileMatchersTest.java [hunk 4]
```diff
@@ -38,6 +38,10 @@
 public class FileMatchersTest
     extends TestSupport
 {
+  private static final String NEW_LINE = System.lineSeparator();
```
Comment (mpiggott) [inline testsupport/src/test/java/org/sonatype/goodies/testsupport/hamcrest/FileMatchersTest.java:41]: These tests didn't work on Windows, needed to test locally

**Review by Blacktiger — APPROVED** [review]

**Pull request merged**